src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/AdditionalConditionWithConfig.kt

```diff
@@ -0,0 +1,33 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model
+
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalCondition
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.AdditionalConditions
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.IAdditionalCondition
+
+data class AdditionalConditionWithConfig(
+  val additionalCondition: AdditionalCondition,
+  val config: IAdditionalCondition,
+)
+
+fun mapConditionsToConfig(licenceConditions: List<AdditionalCondition>, policy: AdditionalConditions): List<AdditionalConditionWithConfig> {
+  val policyAdditionalConditions = policy.ap + policy.pss
+  return licenceConditions.map {
+    val policyCondition = policyAdditionalConditions.find { pc -> it.conditionCode == pc.code }!!
+    AdditionalConditionWithConfig(it, policyCondition)
+  }
+}
+
+fun checkConditionsReadyToSubmit(licenceConditions: List<AdditionalCondition>, policy: AdditionalConditions): Map<String, Boolean> {
+  if (licenceConditions.isEmpty()) { return emptyMap() }
+  val conditionsWithConfig = mapConditionsToConfig(licenceConditions, policy)
+  return conditionsWithConfig.associate {
+    val enteredFields = it.additionalCondition.additionalConditionData.map { data -> data.dataField }
+    val inputEntered =
+      if (!it.config.requiresInput) {
+        true
+      } else {
+        it.config.getConditionInputs()!!.map { input -> input.name }.any { name -> enteredFields.contains(name) }
```

Comment:
is this right? "any input has a data value"? shouldn't it be "all inputs have a corresponding data value " ?

Reply:
Yeah, that's right
Until we have `optional` flags on inputs in the condition config, we don't have a reliable way to determine what's optional and what's not in the API. It's all controlled by the frontend validators. 
Because of that, there are instances where a condition can have some of its inputs populated and not others, and that's valid and should be able to be submitted.
On the flip-side, for conditions that don't have optional inputs, the data is all-or-nothing, you can't submit condition data without submitting all of the required data (due to the frontend validators) meaning that the only time that a condition can have some of its inputs populated and not others are when the others are optional.

I thought adding those optional flags were somewhat out of scope for this ticket, so didn't add them now.